### DISCUSSION TITLE
Python support for edge functions

### DISCUSSION THREAD

**Discussion by Oidlichtnwoada** [supabase/supabase#22944] [Feature Requests]
It would be incredibly helpful to also support Python on the edge, as only with Python, most of the use cases an ordinary backend covers can be handled with supabase as some libraries only support Python. In the future Rust should be added as well or is maybe already supported through WASM. This would cover 99% of all commonly used libraries.

**Comment by yaronhb** [comment]
Yes, this would be very desirable!

In general, it would be nice to have the ability to have other languages.

Another idea to consider is even generic containers that can be invoked in a serverless manner for specific intensive functions [this idea is just a thought that needs to be further developed].

**Comment by tristdrum** [comment]
Viva Python support!

**Comment by augcampos** [comment]
Yes, I got some Firebase python cloud functions I really need to migrate to supabase
Please :100:

**Comment by EugeneLiang** [comment]
Curious if there's any updates? or if it's planned on the roadmap?

**Comment by Zegoverno** [comment]
Any update one this? That would be a great addition...
By now, I am setting a hasura.io API (where they support python) and conecting back to supabase using the wrappers WASM.

**Comment by ogiel** [comment]
I'm trying to convert a PDF to image within edge function, would be so much easier to have Python. Can't get it to work with any TS/JS lib so far.

Here for Python support :D

**Reply by benjaminye** [reply]
I would also like this feature; am too building a PDF to image + parser microservice 😅

**Comment by leonaburime** [comment]
Im shocked this isnt here already.  It would open up a world of Machine Learning and AI applications since Python is so much more mature for this.

**Comment by trancethehuman** [comment]
With all the NLP, ML and LLM stuff I need, a Python Edge Function is going to save my life.

**Comment by leonaburime** [comment]
Is there any way for the community to support this somehow, such as creating a package or something?
Honestly, I'm not the best coder but Id' be willing to at least take a look at what needs to be done as I'm sure many of you would

**Reply by Zegoverno** [reply]
me too here, I would be interested in help, but lack the knowledge to start this from zero.

**Comment by kallebysantos** [comment]
Hi guys, I would like to announce that we're evolving the ML environment for `edge-runtime`. Unfortunately its not python support but already did improve the `onnx` model compatibility and the [Inference API ](https://github.com/supabase/edge-runtime/issues/479) is the next step on it, so it will be possible to get custom ML model support! 😁

**Comment by kasperBengtson** [comment]
I'm just a citizen developer, but I'm thinking: Would it be possible to just use Vercels serverless functions when necessary, if the project is hosted there anyway?

**Comment by bdjafer** [comment]
hi there, any update on that ? is it planned on the roadmap ?

**Comment by ayyshim** [comment]
need python support

**Comment by CatalanCabbage** [comment]
Unfortunately all my existing code is Python + SQLAlchemy ORM. Migrating to Supabase is so hard, since it means rewriting functions code in Typescript :(

Python support please!

**Comment by AC-abatlouni** [comment]
Would like to make supabase a longterm solution for a plethora of projects but I cannot commit without the inclusion of python for developing edge functions. Seems like this would be a win-win for supabase and users/potential users. Concerning there's no affirmation that this is in-development or on the roadmap.

**Comment by sloikodavid** [comment]
boop, this would be amazing for AI startups

**Reply by ashleymajar** [reply]
This, exactly why I need it too

**Comment by moracabanas** [comment]
I want to stop writting full featured FastAPI applications to serve models and domain logic and start teaching my team to write serverless functions triggered by supabase realtime 🙏

**Comment by nichsedge** [comment]
I'm using Python's IMAP and SMTP libraries, and their email parser is incredibly good. I'm trying to convert the functionality to Deno and TypeScript for a Supabase Edge Function, using tools like ChatGPT, Claude, and Gemini. However, none of the results from these tools are as effective as the simple Python script.

**Comment by ms2296** [comment]
is there any update on this? is it part of the roadmap? this would be incredibly useful, similar to how AWS has the AWS Lambda functions.

**Comment by AlbertDoesProgramming** [comment]
+1

**Comment by deepmehta11** [comment]
is there update on python support

**Comment by tec-miguel-cunha** [comment]
Does anyone know anything? @supabase-automated-user

**Comment by Chicoparada** [comment]
Is this on the roadmap? It would be a life saver.

**Comment by thanhson13** [comment]
Supabase edge function's Cloudflare workers wrapper. Until the Python is generally available, we can't have Python runtime
https://developers.cloudflare.com/workers/languages/python/

**Comment by irux** [comment]
Any news?
